Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -12835,8 +12848,324 @@ def pan_os_delete_application_group_command(args):
+
+    Returns:
+        str: a string representing a query with added time filter parameter
+    """
+    if 'time_generated' in query or not last_fetch:
+        return query
+    time_generated = ' and (time_generated geq ' + '\'' + last_fetch.strftime(QUERY_DATE_FORMAT) + '\')'
+    return query + time_generated
+
+
+def add_unique_id_filter_to_query_parameter(query: str, last_id: str) -> str:
+    """append unique id filter parameter to original query parameter.
+    'seqno' is a 64-bit log entry identifier incremented sequentially; each log type has unique number space.
+    by adding this filter which gives us only entries that have larger id, we insure not have duplicates in our request.
+
+    Args:
+        query (str): a string representing a query
+        last_id (str): largest unique log entry id from last fetch (for a specific log type)
+
+    Returns:
+        str: a string representing a query with added time filter parameter
+    """
+    if last_id:
+        last_id_int = arg_to_number(last_id)
+        last_id_int += 1
+        unique_id_filter = ' and (seqno geq ' + '\'' + str(last_id_int) + '\')'
+        return query + unique_id_filter
+    else:
+        return query
+
+
+def fetch_incidents_request(queries_dict: Optional[Dict[str, str]],
+                            max_fetch: int, fetch_start_datetime_dict: Dict[str, datetime], last_id_dict: Dict[str,str]) -> Dict[str, List[Dict[str,Any]]]:
+    """get raw entires of incidents according to provided queries, log types and max_fetch parameters.
+
+    Args:
+        queries_dict (Optional[Dict[str, str]]): chosen log type queries dictionaries
+        max_fetch (int): max incidents per fetch parameter
+        fetch_start_datetime_dict (Dict[str,datetime]): updated last fetch time per log type dictionary
+
+    Returns:
+        Dict[str, List[Dict[str,Any]]]: a dictionary of all fetched raw incidents entries
+    """
+    entries = {}
+    if queries_dict:
+        for log_type, query in queries_dict.items():
+            fetch_start_time = fetch_start_datetime_dict.get(log_type)
+            if fetch_start_time:
+                query = add_time_filter_to_query_parameter(query, fetch_start_time)
+            if id := last_id_dict.get(log_type):
+                query = add_unique_id_filter_to_query_parameter(query, id)
+            response_entries = get_query_entries(log_type, query, max_fetch)
+            entries[log_type] = response_entries
+    return entries
+
+
+def parse_incident_entries(incident_entries: List[Dict[str, Any]]) -> Tuple[str | None ,datetime | None, List[Dict[str, Any]]]:
+    """parses raw incident entries of a specific log type query into basic context incidents.
+    
+    Args:
+        incident_entries (list[dict[str,Any]]): list of dictionaries representing raw incident entries
+
+    Returns:
+        (datetime, List[Dict[str, Any]]): (updated last fetch time, parsed incident list) tuple
```

Comment:
looks like it returns 3 arguments from this function, need to update docstring here